File: src/main/java/com/homechange/api/rest/dto/message/MessageResponseDTO.java
package com.homechange.api.rest.dto.message;

import com.homechange.api.model.Message;

/**
 * Created by Nemanja on 5/22/17.
 *
 * DTO Used for displaying messages
 */
public class MessageResponseDTO {

	private Long messageId;
	private String sentBy;
	private String sentTo;
	private String dateAndTime;
	private String message;

	public MessageResponseDTO() {
	}

	// Contructor that maps Message to this
	public MessageResponseDTO(Message message, String dateAndTime) {
		this.messageId = message.getId();
		this.sentBy = message.getSender().getUsername();
		this.sentTo = message.getRecipient().getUsername();
		this.message = message.getMessage();
		this.dateAndTime = dateAndTime;
	}

	public String getSentBy() {
		return sentBy;
	}

	public void setSentBy(String sentBy) {
		this.sentBy = sentBy;
	}

	public String getSentTo() {
		return sentTo;
	}

	public void setSentTo(String sentTo) {
		this.sentTo = sentTo;
	}

	public String getDateAndTime() {
		return dateAndTime;
	}

	public void setDateAndTime(String dateAndTime) {
		this.dateAndTime = dateAndTime;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}

	public Long getMessageId() {
		return messageId;
	}

	public void setMessageId(Long messageId) {
		this.messageId = messageId;
	}
}

File: src/main/java/com/homechange/api/rest/dto/message/MessageResponseDTO.kt
package com.homechange.api.rest.dto.message

import com.homechange.api.model.Message

/**
 * Created by Nemanja on 5/22/17.
 *
 * DTO Used for displaying messages
 * It's not data class since it has method in it
 */
class MessageResponseDTO {

    var messageId: Long? = null
    var sentBy: String? = null
    var sentTo: String? = null
    var dateAndTime: String? = null
    var message: String? = null

    constructor() {}

    // Constructor that maps Message to this
    constructor(message: Message?, dateAndTime: String?) {
        this.messageId = message?.id
        this.sentBy = message?.sender?.username
        this.sentTo = message?.recipient?.username
        this.message = message?.message
        this.dateAndTime = dateAndTime
    }
}
